lib/private/User/Service/SigninWithEmail.php

```
@@ -0,0 +1,384 @@
+		// Check if current user (active and not in incognito mode)
+		// is an admin
+		$activeUser = $this->userSession->getUser();
+		if ($activeUser !== null) {
+			return $this->groupManager->isAdmin($activeUser->getUID());
+		}
+		// Check if it is triggered from command line
+		if (\OC::$CLI) {
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * @param IUser $user
+	 * @param array $userGroups
+	 * @return array
+	 */
+	public function formatUserForIndex(IUser $user, array $userGroups = null) {
+
+		// TODO: eliminate this encryption specific code below and somehow
+		// hook in additional user info from other apps
+
+		// recovery isn't possible if admin or user has it disabled and encryption
+		// is enabled - so we eliminate the else paths in the conditional tree
+		// below
+		$restorePossible = false;
+
+		if ($this->isEncryptionAppEnabled) {
+			if ($this->isRestoreEnabled) {
+				// check for the users recovery setting
+				$recoveryMode = $this->config->getUserValue($user->getUID(), 'encryption', 'recoveryEnabled', '0');
+				// method call inside empty is possible with PHP 5.5+
+				$recoveryModeEnabled = !empty($recoveryMode);
+				if ($recoveryModeEnabled) {
+					// user also has recovery mode enabled
+					$restorePossible = true;
+				}
+			}
+		} else {
+			// recovery is possible if encryption is disabled (plain files are
+			// available)
+			$restorePossible = true;
+		}
+
+		$subAdminGroups = $this->groupManager->getSubAdmin()->getSubAdminsGroups($user);
+		foreach ($subAdminGroups as $key => $subAdminGroup) {
+			$subAdminGroups[$key] = $subAdminGroup->getGID();
+		}
+
+		$displayName = $user->getEMailAddress();
+		if ($displayName === null) {
+			$displayName = '';
+		}
+
+		$avatarAvailable = false;
+		if ($this->config->getSystemValue('enable_avatars', true) === true) {
+			try {
+				$avatarAvailable = $this->avatarManager->getAvatar($user->getUID())->exists();
+			} catch (\Exception $e) {
+				//No avatar yet
```

Log something here, even if it's just a debug message. If something goes wrong getting the avatar, this will be impossible to detect, debug and fix. Note that you're catching any exception even the `NobodyExpectsTheSpanishInquisitionException` that could happen uploading such file as avatar (it's a joke, but ilustrates why it's a bad idea swallowing exceptions like this)